Update register, getGroupInfo

cdiazucf opened the pull request:
register: added logic to detect wheather user is validated for duplicate login registration
getGroupInfo: returns names instead of only userIDs for secretshopper and group members

cdiazucf committed:
Update register, getGroupInfo

register: added logic to detect wheather user is validated for duplicate login registration
getGroupInfo: returns names instead of only userIDs for secretshopper and group members

gdijkhoffz commented on api.js:
```diff
@@ -202,17 +216,42 @@ exports.setApp = function (app, client)
             const results = await db.collection('Groups').find({ _id:gid}).toArray();
 
+            var index = -1;
+            for(i=0; i <results[0].members.length; i++)
```
Nothing here is wrong, but just a suggestion that can sometimes make code faster: 

When you're accessing a list, in this case "results", several times like results[0] or even results[0].members. It can help to save it in another variable so you don't keep having to keep indexing the list every time. Similarly with the length function, it's easier to calculate the length before the loop, so we don't have to keep calculating the same value. For example here, you could do the following:

```
var members = results[0].members;
var len = members.length;

for (i = 0; i < len ; i++) {
  if (members[i] == userId)
    index = i;
}
```

gdijkhoffz reviewed (approved):
Looks good to me 👍

cdiazucf committed:
two new endpoints, refactored for efficiency

added deleteUser, forgotPasswordRequest, forgotPasswordReset, getUser, updateUser
updated variable declarations for efficienciy
